### ISSUE TITLE
no-unused-variable false positive

### ISSUE THREAD

**Issue by kuhnroyal** [palantir/tslint#3484]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.1
- __Running TSLint via__: nodejs, tslint-loader

#### TypeScript code being linted

```ts
import { Foo } from "src/foo";
import {
  IIdentifiable,
  IQuery
} from "src/resource";

// either of the following fails
export interface FooAction extends FluxStandardAction<Foo[] | Foo, {
  id?: IIdentifiable;
  query?: IQuery;
}> {}

export type FooActionType = FluxStandardAction<Foo[] | Foo, {
  id?: IIdentifiable;
  query?: IQuery;
}> ;

// 
```

with `tslint.json` configuration:

```json
    "no-unused-variable": [
      true,
      "check-parameters",
      {
        "ignore-pattern": "^_"
      }
    ],
```

#### Actual behavior
Fixer removes the imports for `IIdentifiable` and `IQuery`.

#### Expected behavior
Fixer should keep the imports.

**Comment by ajafff** [comment]
This is likely a duplicate of #3463
